internals/secrethub/credential_disable_test.go

```diff
@@ -0,0 +1,141 @@
+
+	"github.com/secrethub/secrethub-cli/internals/cli/ui/fakeui"
+	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/pkg/secrethub"
+	"github.com/secrethub/secrethub-go/pkg/secrethub/fakeclient"
+)
+
+func TestCredentialDisableCommand_Run(t *testing.T) {
+
+	var testError = errors.New("test")
+
+	testCases := map[string]struct {
+		cmd            CredentialDisableCommand
+		force          bool
+		promptOut      string
+		out            string
+		in             string
+		disableError   error
+		newClientError error
+		promptError    error
+		err            error
+	}{
+		"fail-force-no-credential": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "",
+				force:       true,
+			},
+			err: errors.New("fingerprint argument must be set when using --force"),
+		},
+		"fail-prompt-error": {
+			promptError: testError,
+			err:         testError,
+		},
+		"fail-empty-credential-3-times": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "",
+				force:       false,
+			},
+			promptOut: "What is the fingerprint of the credential you want to disable? \n" +
+				"Invalid input: fingerprint is invalid (api.invalid_fingerprint) \n" +
+				"Please try again.\n" +
+				"What is the fingerprint of the credential you want to disable? \n" +
+				"Invalid input: fingerprint is invalid (api.invalid_fingerprint) \n" +
+				"Please try again.\n" +
+				"What is the fingerprint of the credential you want to disable? \n" +
+				"Invalid input: fingerprint is invalid (api.invalid_fingerprint) \n",
+			err: api.ErrInvalidFingerprint,
+		},
+		"fail-abort": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "62542d734d7f3627",
+				force:       false,
+			},
+			promptOut: "Are you sure you want to disable the credential with fingerprint 62542d734d7f3627? [y/N]: ",
+			in:        "n",
+			out:       "A disabled credential can no longer be used to access SecretHub. This process can currently not be reversed.\nAborting.\n",
+		},
+		"fail-client-error": {
+			newClientError: testError,
+			err:            testError,
+		},
+		"fail-force-wrong-credential": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "BillyBoy",
+				force:       true,
+			},
+			err: api.ErrInvalidFingerprint,
+		},
+		"fail-too-short-fingerprint": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "6254",
+				force:       true,
+			},
+			err: api.ErrTooShortFingerprint,
+		},
+		"succeed-force": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "62542d734d7f3627",
+				force:       true,
+			},
+			out: "A disabled credential can no longer be used to access SecretHub. " +
+				"This process can currently not be reversed.\nCredential disabled.\n",
+		},
+		"succeed-no-force": {
+			cmd: CredentialDisableCommand{
+				fingerprint: "62542d734d7f3627",
+				force:       false,
+			},
+			promptOut: "Are you sure you want to disable the credential with fingerprint 62542d734d7f3627? [y/N]: ",
```

Comment:
This fingerprint shows up in several places in this test function, therefore it would be a good idea to define it as a constant.

If it is used in a sentence you can use the [`fmt.Sprintf`](https://golang.org/pkg/fmt/#Sprintf) function to construct it.

Reply:
This should now be solved! :D